server/api/availability_handler.py

```diff
@@ -10,10 +10,12 @@
 
 availability_handler = Blueprint('availability_handler', __name__)
 
-@availability_handler.route("/availability/<int:id>", methods=["GET"])
+
+@availability_handler.route("/availability/<url>", methods=["GET"])
 @check_token
-def get_calendar_availability(id):
-    user = Users.query.filter_by(id=id).first()
+def get_calendar_availability(url):
+    uid = EventTypes.query.filter_by(url=url).first().user_id
+    user = Users.query.filter_by(id=uid).first()
```

Comment:
Need to add some validation here:

1. Handle event urls that don't exist, in the case of someone randomly enters calendar/ RANDOM-URL.
2. Should we include the user's url as part of the url as well? i.e. calendar/user_url/event-type_url. This way we can be sure that we are looking at the correct person's calendar.

Reply:
@fangyu3 I've fixed the two url issues above and make username and other text dynamic. Please review once again, thanks